engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/RService.java

```diff
@@ -24,8 +23,10 @@
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
 
 @Slf4j
+@Service
```

Comment:
You're right. I might have missed this was an abstract class. I will add the `@Service` annotation to `DockerService `class since `Autowired `members should be defined in a Spring Bean.

Reply:
This change has broken the application:
```
Caused by: java.lang.IllegalStateException: Duplicate key com.odysseusinc.arachne.executionengine.execution.r.DockerService@58cf8f94
	at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133) ~[na:1.8.0_392]
	at java.util.HashMap.merge(HashMap.java:1255) ~[na:1.8.0_392]
	at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320) ~[na:1.8.0_392]
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169) ~[na:1.8.0_392]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[na:1.8.0_392]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[na:1.8.0_392]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[na:1.8.0_392]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[na:1.8.0_392]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:1.8.0_392]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[na:1.8.0_392]
	at com.odysseusinc.arachne.executionengine.execution.AnalysisService.<init>(AnalysisService.java:92) ~[30d8ab8a-5585-4ee9-959c-26a69c5d5890/:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.8.0_392]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:1.8.0_392]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_392]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[na:1.8.0_392]
```

Reply:
I will remove this change since looks like it was already registered as a Bean.